### PR TITLE
[KYUUBI #871] Change module name kyuubi-main to kyuubi-server

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

As we will release soon, it's better to format the module name and artifactId.

The name of kyuubi-main sounds like a bootstrap of Kyuubi but not a server.

### _How was this patch tested?_
Pass CI

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn @pan3793 @turboFei

**Review by yaooqinn — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,7 @@
         <module>kyuubi-ctl</module>
         <module>kyuubi-ha</module>
         <module>kyuubi-hive-jdbc</module>
-        <module>kyuubi-main</module>
+        <module>kyuubi-server</module>
```
Comment (pan3793) [inline pom.xml:87]: move to next line

File: kyuubi-server/pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
     </parent>
     <modelVersion>4.0.0</modelVersion>
 
-    <artifactId>kyuubi-main</artifactId>
+    <artifactId>kyuubi-server</artifactId>
     <name>Kyuubi Project Server Main</name>
```
Comment (pan3793) [inline kyuubi-server/pom.xml:30]: `<name>Kyuubi Project Server Main</name>` => `<name>Kyuubi Project Server</name>`

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master for v1.3.0

**Pull request closed without merging**